numpyro/util.py

```diff
@@ -171,3 +172,27 @@ def decorator(destin_class):
         return destin_class
 
     return decorator
+
+
+pytree_metadata = namedtuple('pytree_metadata', ['flat', 'shape', 'size', 'dtype'])
+
+
+def _ravel_list(*leaves):
+    leaves_metadata = tree_map(lambda l: pytree_metadata(np.ravel(l), np.shape(l), np.size(l), lax.dtype(l)),
+                               leaves)
+    leaves_idx = np.cumsum(np.array((0,) + tuple(d.size for d in leaves_metadata)))
+
+    def unravel_list(arr): return [np.reshape(lax.dynamic_slice_in_dim(arr, leaves_idx[i], m.size),
```

Comment:
nit: I think it is easier to read if we start with new line (or use `unravel_list = lambda arr: ...`)

Reply:
Done